modules/serverless-typescript-context-object-reference.adoc

```diff
@@ -0,0 +1,130 @@
+[id="serverless-typescript-context-object-reference_{context}"]
+= TypeScript context object reference
+
+The `context` object has several properties that can be accessed by the function developer.
+
+[id="serverless-typescript-context-object-reference-log_{context}"]
+== log
+
+Provides a logging object that can be used to write output to the cluster logs. The log adheres to the link:https://getpino.io/#/docs/api[Pino logging API].
+
+.Example log
+[source,javascript]
+----
+function handle(context: Context) Record<string,string> {
+  context.log.info(“Processing customer”);
+  return 'OK';
+}
+----
+
+You can access the function by using the `kn func emit` command to invoke it:
+
+.Example command
+[source,terminal]
+----
+$ kn func emit --sink local
+----
+
+.Example output
+[source,terminal]
+----
+{"level":30,"time":1604511655265,"pid":3430203,"hostname":"localhost.localdomain","reqId":1,"msg":"Processing customer"}
+----
+
+You can change the log level to one of `fatal`, `error`, `warn`, `info`, `debug`, `trace` or `silent`. To do that, change the value of `logLevel` to one of these values using the `config` command. You can also temporarily override the value of `logLevel` by assigning one of the log level values to the environment variable `FUNC_LOG_LEVEL`.
+
+[id="serverless-typescript-context-object-reference-query_{context}"]
+== query
+
+Returns the query string for the request, if any, as key-value pairs. These attributes are also found on the context object itself.
+
+.Example query
+[source,javascript]
+----
+function handle(context: Context) Record<string,string> {
+  // Log the 'name' query parameter
+  context.log.info(context.query.name);
+  // Query parameters are also attached to the context
+  context.log.info(context.name);
+}
+----
+
+You can access the function by using the `kn func emit` command to invoke it:
+
+.Example command
+[source,terminal]
+----
+$ kn func emit --sink local --data '{"name": "tiger"}'
```

Comment:
Here again is the issue of `--sink local`. It's again possible that the dev has already deployed and come back to the reference doc. There's probably a good way to deal with this in one place for all of these examples - maybe? It's not a blocker for me, but just something to give some thought to.